main.cpp

```diff
@@ -115,17 +143,20 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T typeA( 5, "typeA");                                             //6
+    T typeB( 2, "typeB");                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    Values f;                                            //7
+    auto* smaller = f.compare(&typeA, &typeB);                              //8
+    if(smaller != nullptr)
+        std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    else
+        std::cout << "Warning: smaller pointer is nullptr" << std::endl; //9
```

Comment:
I have revised instruction 9. 
please revise this `else` statement as described by the new instructions
- [ ] revise